pkg/driver/node.go

```diff
@@ -71,30 +73,57 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		return nil, status.Error(codes.InvalidArgument, "Volume capability not supported")
 	}
 
+	ok := d.inFlight.Insert(req)
+	if !ok {
+		msg := fmt.Sprintf("request to stage volume=%+v is already in process: formatting volume", volumeID)
+		return nil, status.Error(codes.Internal, msg)
+	}
+	defer func() {
```

Comment:
Just to confirm this function is synchronous and some call below will block?

Reply:
Yes, namely FormatAndMount.